example/ck_tile/35_batched_transpose/batched_transpose.cpp

```diff
@@ -0,0 +1,276 @@
+
+    ck_tile::DeviceMem x_dev(x_host.get_element_space_size_in_bytes());
+    ck_tile::DeviceMem y_dev(y_host.get_element_space_size_in_bytes());
+
+    x_dev.ToDevice(x_host.data());
+
+    transpose_kernel_param_t kparam;
+    for(auto iter_kparam : get_transpose_all_kernel(prec))
+    {
+        bool is_valid = is_kernel_valid(N, C, H, W, &iter_kparam, layout_in);
+        if(is_valid)
+        {
+            kparam = iter_kparam;
+            break;
+        }
+    }
+
+    auto trait = batched_transpose_trait{prec, layout_in};
+
+    uint32_t height      = nchw2nhwc ? C : H * W;
+    uint32_t width       = nchw2nhwc ? H * W : C;
+    uint32_t dim_block_h = (height + kparam.tile_y - 1) / kparam.tile_y;
+    uint32_t dim_block_w = (width + kparam.tile_x - 1) / kparam.tile_x;
+    uint32_t dim_stride  = height * width;
+    uint32_t dim_blocks  = dim_block_h * dim_block_w;
+
+    batched_transpose_kargs karg = [&]() {
+        batched_transpose_kargs a_;
+        a_.p_input     = x_dev.GetDeviceBuffer();
+        a_.p_output    = y_dev.GetDeviceBuffer();
+        a_.batch       = N;
+        a_.height      = height;
+        a_.width       = width;
+        a_.dim_blocks  = dim_blocks;
+        a_.dim_stride  = dim_stride;
+        a_.dim_block_h = dim_block_h;
+        a_.dim_block_w = dim_block_w;
+        return a_;
+    }();
+
+    ck_tile::stream_config sc{nullptr, true};
+
+    auto ms = batched_transpose(trait, karg, sc);
+    printf("[%s]N:%d, C:%d, H:%d, W:%d, layout_in:%s, %f\n",
+           prec.c_str(),
+           N,
+           C,
+           H,
+           W,
+           layout_in.c_str(),
+           ms);
+    if(ms < 0)
+        printf("not supported\n");
+    fflush(stdout);
+
+    if(ms < 0)
+    {
+        return false;
+    }
+
+    y_dev.FromDevice(y_host.data());
+
+    bool rtn = true;
+    if(validate)
+    {
+        // this host buffer will not copy to GPU, so no need use stride
+        ck_tile::HostTensor<Type> y_ref(
+            {dim_out[0], dim_out[1], dim_out[2], dim_out[3]},
+            {stride_dim_out[0], stride_dim_out[1], stride_dim_out[2], stride_dim_out[3]});
+
+        // dump_host_tensor_4d(x_host);
+        ck_tile::reference_batched_transpose<Type>(x_host, y_ref, layout_in, layout_out);
+        // printf("y reference:\n");
+        // dump_host_tensor_4d(y_ref);
+
+        // printf("y host:\n");
+        // dump_host_tensor_4d(y_host);
+
+        auto [rtol, atol] = get_elimit<Type>("");
+
+        rtn &= ck_tile::check_err(
+            y_host, y_ref, std::string("y Error: Incorrect results!"), rtol, atol);
+    }
+    printf("valid:%s\n", rtn ? "y" : "n");
+    fflush(stdout);
+    return rtn;
+}
+
+int main(int argc, char** argv)
+{
+    auto [result, args] = create_args(argc, argv);
+    if(!result)
+        return -1;
+    std::string prec = args.get_str("pr");
+
+    bool r = true;
+    if(prec.compare("fp32") == 0)
+    {
+        r &= test_batched_transpose<float>(args);
```

Comment:
We are doing the example run instead of the test run. test_batched_transpose seems inappropriate. I suggest to change the function name to run_batched_transpose. WDYT?